lib/matplotlib/axes/_axes.py

```diff
@@ -6637,7 +6637,16 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
             bin_range = self.convert_xunits(bin_range)
 
         # Check whether bins or range are given explicitly.
-        binsgiven = np.iterable(bins) or bin_range is not None
+        if bins is None:
```

Comment:
The whole thing is just `bins_array_given = np.ndim(bins) == 1`? (perhaps with additional comments indicating that this excludes str and None).